### PR TITLE
diagnose: Add seal transit tls check

### PR DESCRIPTION
**Pull request by hghaf099**

### CODE REVIEW

File: command/operator_diagnose_test.go
Comment (hghaf099) [inline command/operator_diagnose_test.go:324]: I am still working on this, it panics!

Comment (hghaf099) [reply command/operator_diagnose_test.go]: hmm, bad path to the config file cause vault to panic! Wondering if that should be resolved!

File: command/operator_diagnose.go
Comment (hghaf099) [inline command/operator_diagnose.go:395]: warn if any of the cert not exist.

File: command/operator_diagnose.go
Comment (hghaf099) [inline command/operator_diagnose.go:385]: just compare to "true"

File: command/operator_diagnose.go
Comment (hghaf099) [inline command/operator_diagnose.go:410]: warn if not exist

File: command/operator_diagnose.go
Comment (hghaf099) [inline command/operator_diagnose.go:385]: I have not returned anything here. This has made the output of diagnose to misbehave. In particular, "warn" message has indented a little and new empty line was printed above it. 
here is a snapshot:

Screen Shot 2021-06-23 at 4 54 53 PM

Comment (hghaf099) [reply command/operator_diagnose.go:385]: Also, it seems that the name of the diagnose test is also not printed, which is seal-transit-tls-checks.
Here is a snapshot when I do return nil right after that warning
Screen Shot 2021-06-23 at 4 57 54 PM

**Review by HridoyRoy — COMMENTED** [review]
This PR is currently blocked on https://github.com/hashicorp/vault/pull/11910 . We can only verify correctness of the output in this PR after removing the bugs fixed in the linked PR and ensuring that the fix carried over. If it did not, we'd need to address the warns first. I'll update https://hashicorp.atlassian.net/browse/VAULT-2736 and https://hashicorp.atlassian.net/browse/VAULT-1897 indicating this. Thanks!

File: command/operator_diagnose.go
Comment (HridoyRoy) [reply command/operator_diagnose.go:399]: Hopefully this is fixed after merging main?

File: command/operator_diagnose.go
Comment (HridoyRoy) [inline command/operator_diagnose.go:399]: No need to warn if `tls_skip_verify` doesn't exist. In fact, in most cases it should not exist.

File: command/operator_diagnose.go
Comment (HridoyRoy) [inline command/operator_diagnose.go:410]:
```suggestion
                                        diagnose.Advise(ctx, "If TLS client verification is required by the vault server, please set the value of `tls_client_cert` in the corresponding config stanza. Please refer to our documentation for more details: https://www.vaultproject.io/docs/configuration/seal/transit .")
					diagnose.Warn(ctx, "TLS Client Certificate is missing from a seal stanza of the vault configuration.")
```

File: command/operator_diagnose.go
Comment (HridoyRoy) [inline command/operator_diagnose.go:415]:
```suggestion
                                        diagnose.Advise(ctx, "If TLS client verification is required by the vault server, please set the value of `tls_client_key` in the corresponding config stanza. Please refer to our documentation for more details: https://www.vaultproject.io/docs/configuration/seal/transit .")
					diagnose.Warn(ctx, "TLS Client Key is missing from a seal stanza of the vault configuration.")
```

File: command/operator_diagnose.go
Comment (HridoyRoy) [inline command/operator_diagnose.go:426]:
```suggestion
					diagnose.Advise(ctx, "If TLS is required by the vault server, please set the value of `tls_ca_cert` in the corresponding config stanza. Please refer to our documentation for more details: https://www.vaultproject.io/docs/configuration/seal/transit .")
					diagnose.Warn(ctx, "TLS Client CA Certificate is missing from a seal stanza of the vault configuration.")
```

File: command/operator_diagnose.go
Comment (HridoyRoy) [inline command/operator_diagnose.go:420]: Might be helpful to wrap this with a message describing that the TLS error being thrown is explicitly for the client certificate and key?

File: command/operator_diagnose.go
Comment (HridoyRoy) [inline command/operator_diagnose.go:431]: Same here -- it would be great to wrap this error in a message clarifying that this TLS error was for the CA cert file for one of the seals.

**Review by HridoyRoy — CHANGES_REQUESTED** [review]
Just a few small comments on error messages. Looks great overall!

File: command/operator_diagnose.go
Comment (hghaf099) [reply command/operator_diagnose.go:399]: Done

**Review by HridoyRoy — APPROVED** [review]

**Pull request merged**
